### PR TITLE
Fix invalid type URL if submodule is main entry

### LINKED ISSUE #578: Incorrect `x-typescript-types` header when main entry is fetched directly
## Failing module

- **GitHub**: https://github.com/KaTeX/KaTeX
- **npm**: https://npmjs.com/package/katex

```js
import * as katex from "https://esm.sh/katex@0.16.4/dist/katex.mjs"
```

## Error message

Not an error per se, but deno is unable to find the correct typings. See discussion at https://github.com/denoland/deno/issues/18583

The problem is that deno resolves the main package entry itself and fetches `https://esm.sh/katex@0.16.4/dist/katex.mjs` directly instead of `https://esm.sh/katex@0.16.4`. The latter has the correct `x-typescript-types` header, but the former points to an invalid one. Given that the former is an exact match of a main entry point I was expecting the main entry point to be resolved against `@types/katex`.

| URL | x-typescript-types | Status |
|---|---|---|
| https://esm.sh/katex@0.16.4/dist/katex.mjs | https://esm.sh/v113/@types/katex@~0.16/dist/katex.mjs~.d.ts | wrong |
| https://esm.sh/katex@0.16.4 | https://esm.sh/v113/@types/katex@~0.16/index.d.ts | correct |

`katax` package.json:

```json
{
  "name": "katex",
  "version": "0.16.4",
  "main": "dist/katex.js",
  "exports": {
    ".": {
      "require": "./dist/katex.js",
      "import": "./dist/katex.mjs" // <-- This file is fetched directly by deno
    },
    // ...
  }
}
```

`@types/katax` package.json:

```json
{
    "name": "@types/katex",
    "version": "0.16.0",
    "main": "",
    "types": "index.d.ts",
    //...
}
```

## Additional info

- **esm.sh version**: latest - whatever is deployed on https://esm.sh
- **Deno version**: 1.32.3
- Related issue: https://github.com/denoland/deno/issues/18583

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Fixes #578

### CODE REVIEW

File: test/issue-578/issue-578.test.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { assertEquals } from "https://deno.land/std@0.178.0/testing/asserts.ts";
+
+Deno.test("issue #478", async () => {
```
Comment (ije) [inline test/issue-578/issue-578.test.ts:3]: typo here it should be "issue #578", doesn't matter if you don't want to change haha

**Review by ije — APPROVED** [review]
Looks so great to me! thanks @marvinhagemeister

File: test/issue-578/issue-578.test.ts (hunk 1 above)
Comment (marvinhagemeister) [reply test/issue-578/issue-578.test.ts]: Whoops, good catch! That's an obvious copy and paste error on part. Corrected the reference to point to the right issue number. Thanks for reviewing!

**Review by ije — APPROVED** [review]
LGTM👍

**Pull request merged**